services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/principal/PrincipalManagerImplUnitTest.java

```diff
@@ -285,78 +282,100 @@ public void testValidateNewAccountTokenBadTimestamp() {
 	public void testValidateNOTtooOLDTimestamp() {
 		Date notOutOfDate = new Date(System.currentTimeMillis()+23*3600*1000L);
 		String token = PrincipalManagerImpl.createTokenForNewAccount(user, now);
-		PrincipalManagerImpl.validateNewAccountToken(token, notOutOfDate);
+		PrincipalManagerImpl.validateEmailToken(token, notOutOfDate);
 	}
 	
 	// token is not OK 25 hours from now
 	@Test(expected=IllegalArgumentException.class)
 	public void testValidateOLDTimestamp() {
 		Date outOfDate = new Date(System.currentTimeMillis()+25*3600*1000L);
 		String token = PrincipalManagerImpl.createTokenForNewAccount(user, now);
-		PrincipalManagerImpl.validateNewAccountToken(token, outOfDate);
+		PrincipalManagerImpl.validateEmailToken(token, outOfDate);
 	}
 	
 	@Test(expected=IllegalArgumentException.class)
 	public void testValidateNewAccountTokenInvalidToken() {
 		testReplacedParamValidateNewAccountToken("mac", "invalid-mac");
 	}
+
 	@Test
 	public void testNewAccountEmailValidationHappyPath() throws Exception {
 		when(mockPrincipalAliasDAO.isAliasAvailable(EMAIL)).thenReturn(true);
-		manager.newAccountEmailValidation(user, "https://www.synapse.org?");
+		manager.newAccountEmailValidation(user, PORTAL_ENDPOINT);
 		ArgumentCaptor<SendRawEmailRequest> argument = ArgumentCaptor.forClass(SendRawEmailRequest.class);
 		verify(mockSynapseEmailService).sendRawEmail(argument.capture());
 		SendRawEmailRequest emailRequest =  argument.getValue();
 		assertEquals(Collections.singletonList(EMAIL), emailRequest.getDestinations());
 		MimeMessage mimeMessage = new MimeMessage(Session.getDefaultInstance(new Properties()),
 				new ByteArrayInputStream(emailRequest.getRawMessage().getData().array()));
 		String body = (String)((MimeMultipart) mimeMessage.getContent()).getBodyPart(0).getContent();
-		assertEquals("Welcome to SYNAPSE!", mimeMessage.getSubject());
+		assertNotNull(mimeMessage.getSubject());
 		// check that all template fields have been replaced
 		assertTrue(body.indexOf("#")<0);
-		assertTrue(body.indexOf(FIRST_NAME)>=0); 
-		// check that user's name appears
-		assertTrue(body.indexOf(LAST_NAME)>=0); 
 		// check that token appears
-		assertTrue(body.indexOf("https://www.synapse.org?")>=0); 
-		assertTrue(body.indexOf("firstname=foo&lastname=bar&email=foo%40bar.com")>=0);
+		assertTrue(body.indexOf(PORTAL_ENDPOINT)>=0);
+		assertTrue(body.lastIndexOf(PORTAL_ENDPOINT) < body.length() - 1);

```

Comment:
body.lastIndexOf("gobbledygook")< ... would pass because if a substring is not in the parent string then the function returns -1.